Improve DateBox demo - formatting description (#501)
DevExpress/devextreme-demos: JSDemos/Demos/DateBox/Formatting/description.md

> To customize the **DateBox** format, use the **displayFormat** property. This demo illustrates the following use cases of this property:

Reviewer: ```suggestion
To customize the **DateBox** format, use the [displayFormat](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#displayFormat) property. This demo illustrates the following use cases of this property:

```

- **Locale-dependent format**     

> This is the default formatting behavior dependent on the User's [locale](/Documentation/Guide/Common/Localization/#Localize_Dates_Numbers_and_Currencies).    

Reviewer: - It is not the behavior, it is the format that depends on the user's locale.
- "formatting" - unnecessary -ing
- Two thoughts in a single sentence: 1) the format depends on the locale; 2) this is the default behavior.
- The locale link doesn't full match its name. I would expect that the page that you linked to describes what a user's locale is. I suggest adding another sentence for this link.

```suggestion
The DateBox formats its value according to the user's locale. This is the default behavior. Refer to the following help topic for more information: [Localize Dates, Numbers, and Currencies](/Documentation/Guide/Common/Localization/#Localize_Dates_Numbers_and_Currencies)
```

- **Built-in predefined format**    

> DevExtreme supports predefined [formats](https://js.devexpress.com/Documentation/ApiReference/Common/Object_Structures/format/#type). Set the **displayFormat** property to the respective format.    

Reviewer: ```suggestion
DevExtreme supports [predefined formats](/Documentation/ApiReference/Common/Object_Structures/format/#type). Set the **displayFormat** property to the respective format.    
```

- **LDML pattern**    

> Set the **displayFormat** property to a custom format string using [LDML](/Documentation/Guide/Common/Value_Formatting/#Format_Widget_Values/Custom_Format_String) patterns.    

Reviewer: - "using" - what uses what?
- I would add a sentence about the nature of LDML patterns.
- The same note about the link that doesn't fully match its name.

```suggestion
An LDML pattern consists of wildcard characters and characters displayed as is. Set the **displayFormat** property to a string that specifies an LDML pattern. Refer to the following help topic for information about supported format characters: [Custom Format String](/Documentation/Guide/Common/Value_Formatting/#Format_Widget_Values/Custom_Format_String)    
```

- **Format with literal characters**    

> The **displayFormat** property can contain literal characters combined with **LDML** characters. Wrap the literal characters in quotation marks.

Reviewer: > The **displayFormat** property can contain literal characters combined with **LDML** characters.

It is better to say what to do with the displayFormat property and then add that the string can contain LDML characters:

```suggestion
Specify the **displayFormat** property as a string that contains literal characters. This string can also contain LDML format characters. Wrap the literal characters in quotation marks.
```

Reviewer 2: As I see, the previous example already contains "literal" characters. Besides the commas and spaces, there's the word "of". So I think it's important to note here that quotes are needed because some of the characters can be interpreted as wildcards. For example, if you remove quotes that surround the word "Year", then "a" will be interpreted as "AM/PM" placeholder.

> Refer to the [displayFormat](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#displayFormat) description for additional information.

Reviewer: It is better to send the reader to the Value Formatting article.

```suggestion
To get more information about value formatting, refer to the [Value Formatting](/Documentation/Guide/Common/Value_Formatting/) article.
```

> Refer to the [displayFormat](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#displayFormat) description for additional information.

Reviewer: We forgot to mention `useMaskBehavior`. Let's add information about it as the last paragraph.

> To customize how a **DateBox** displays dates, use the [displayFormat](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#displayFormat) property. This demo illustrates the following use cases:

Reviewer 3: ```suggestion
To change how a **DateBox** displays dates, use the [displayFormat](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#displayFormat) property. This demo illustrates the following examples:
```

> To customize the **DateBox** format, use the [displayFormat](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#displayFormat) property. This demo illustrates the following use cases of this property:

Reviewer 2: ```suggestion
To customize how a **DateBox** displays dates, use the [displayFormat](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#displayFormat) property. This demo illustrates the following use cases:
```

- **Locale-dependent format**     

> The DateBox formats its value according to the user's locale. This is the default behavior. Refer to the following help topic for more information: [Localize Dates, Numbers, and Currencies](/Documentation/Guide/Common/Localization/#Localize_Dates_Numbers_and_Currencies)

Reviewer 2: I would link all these items to property values. So in this case, what should I use as the "displayFormat" property value. Is that an empty string, or something else? 

Let's say I want a switch in my application that changes between locale-dependent and some predefined format. When I revert to locale-dependent, what value should I use?

Author: As far as I understand the locale-dependent format (and according to the demo), the "displayFormat'' property is not defined as the DateBox automatically detects the user's locale and formats the date/time accordingly.  So in the above-mentioned scenario, we would need to either define the "displayFormat" property or remove it altogether.

> - **Built-in predefined format**    

Reviewer 2: ```suggestion
- **Built-in predefined formats**    
```

> The DateBox formats its value according to the user's locale. This is the default behavior. You don't need to define the **displayFormat** property. Refer to the following help topic for more information: [Localize Dates, Numbers, and Currencies](/Documentation/Guide/Common/Localization/#Localize_Dates_Numbers_and_Currencies)

Reviewer 3: ```suggestion
The DateBox formats its value according to the user's locale. This is the default format and you do not need to define the **displayFormat** property. Refer to the following help topic for more information: [Localize Dates, Numbers, and Currencies](/Documentation/Guide/Common/Localization/#Localize_Dates_Numbers_and_Currencies)
```

> DevExtreme supports [predefined formats](/Documentation/ApiReference/Common/Object_Structures/format/#type). Set the **displayFormat** property to the respective format.    

Reviewer 2: ```suggestion
DevExtreme supports a number of [predefined formats](/Documentation/ApiReference/Common/Object_Structures/format/#type). Those are simple strings or shortcuts that define commonly used date formats so that you don't need to spell out a complex expression. Review the linked document for the list of values you can assign to **displayFormat**.  This demo uses the "shortdate" format. 
```

> DevExtreme supports a number of [predefined formats](/Documentation/ApiReference/Common/Object_Structures/format/#type). Those are simple strings or shortcuts that define commonly used date formats so that you don't need to spell out a complex expression. Review the linked document for the list of values you can assign to **displayFormat**.  This demo uses the "shortdate" format. 

Reviewer 3: ```suggestion
DevExtreme supports various [predefined formats](/Documentation/ApiReference/Common/Object_Structures/format/#type). You can use simple strings or shortcuts that define commonly used date formats instead of complex expressions. Review the linked document for a list of values you can assign to **displayFormat**.  This demo uses the "shortdate" format. 
```

- **LDML pattern**    

> An LDML pattern consists of wildcard characters and characters displayed as is. Set the **displayFormat** property to a string that specifies an LDML pattern. Refer to the following help topic for information about supported format characters: [Custom Format String](/Documentation/Guide/Common/Value_Formatting/#Format_Widget_Values/Custom_Format_String)    

Reviewer 2: ```suggestion
Use an LDML pattern to construct a custom date-time format string. This demo sets **displayFormat** property to "EEEE, d of MMM, yyyy HH:mm" - a combination of day of the week, day number, month, year, hour, and minute. You can see the formatted value in the editor. Refer to the following help topic for information about supported format characters: [Custom Format String](/Documentation/Guide/Common/Value_Formatting/#Format_Widget_Values/Custom_Format_String)    
```

> Use an LDML pattern to construct a custom date-time format string. This demo sets **displayFormat** property to "EEEE, d of MMM, yyyy HH:mm" - a combination of day of the week, day number, month, year, hour, and minute. You can see the formatted value in the editor. Refer to the following help topic for information about supported format characters: [Custom Format String](/Documentation/Guide/Common/Value_Formatting/#Format_Widget_Values/Custom_Format_String)    

Reviewer 3: ```suggestion
Use an LDML pattern to construct a custom date-time format string. This demo sets **displayFormat** property to "EEEE, d of MMM, yyyy HH:mm" - a combination of the day of the week, day number, month, year, hour, and minute. The editor displays the formatted value. Refer to the following help topic for information about supported format characters: [Custom Format String](/Documentation/Guide/Common/Value_Formatting/#Format_Widget_Values/Custom_Format_String)    
```

- **Format with literal characters**    

> Specify the **displayFormat** property as a string that contains literal characters. This string can also contain LDML format characters. Wrap any characters that are not a part of the LDML pattern in quotation marks. Otherwise they may be interpreted as wildcards. In this example, if you remove quotation marks from the word "Year", the letter "a" will be interpreted as "AM/PM" placeholder.

Reviewer 3: ```suggestion
Specify the **displayFormat** property as a string that contains literal and LDML characters. Wrap any characters that are not part of the LDML pattern in quotation marks. Otherwise, they can be interpreted as wildcards. In this example, if you remove quotation marks from the word "Year", the letter "a" is interpreted as an "AM/PM" placeholder.
```

> Note that the **DateBox** supports input masks. Use them to ensure that the input value format is consistent with **displayFormat**. To enable this behavior, set the [useMaskBehavior](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#useMaskBehavior) value to *"true"*.

Reviewer: ```suggestion
Note that the **DateBox** supports input masks. Use them to ensure that the input value format is consistent with **displayFormat**. To enable this behavior, set the [useMaskBehavior](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#useMaskBehavior) value to **true**.
```

> Note that the **DateBox** supports input masks. Use them to ensure that the input value format is consistent with **displayFormat**. To enable this behavior, set the [useMaskBehavior](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#useMaskBehavior) value to *"true"*.

Reviewer 3: ```suggestion
You can also use an input mask in the **DateBox**. Input masks ensure that the input value format is consistent with **displayFormat**. To do this, set the [useMaskBehavior](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#useMaskBehavior) value to *"true"*.
```

Author: With all due respect, in my opinion, this is an unnecessary correction as it doesn't add any value, better inform users, improve grammar, etc. Moreover, using "input mask" in two consecutive sentences makes the text repetitive, which doesn't seem like an improvement over the previous version.  "You can also use input mask in....Input masks.... input....".

Would you care to elaborate on the reasoning behind these corrections?

Reviewer 3: - There is no need to add 'Note this' or 'Note that'. If the DateBox allows you to use input masks, why not just state this?
-  'supports input masks' means what? You can use input masks in the DateBox? If that is the case, why not just say that?
- It is almost always better to say what **this/them/that** refers to
- I agree, an argument could be made to remove **input** from  'the input value format'

You can also use an input mask in the **DateBox**. Input masks ensure the entered value matches the **displayFormat**. To do this, set the [useMaskBehavior](/Documentation/ApiReference/UI_Widgets/dxDateBox/Configuration/#useMaskBehavior) value to *"true"*.

Author: Added this suggestion to the latest commit

> To get more information about value formatting, refer to the [Value Formatting](/Documentation/Guide/Common/Value_Formatting/) article.

Reviewer 3: ```suggestion
For more information about value formatting, refer to the [Value Formatting](/Documentation/Guide/Common/Value_Formatting/) article.
```
